tests/unit_tests/pyanaconda_tests/modules/subscription/test_system_purpose.py

```diff
@@ -0,0 +1,206 @@
+#
+# Copyright (C) 2021  Red Hat, Inc.
+#
+# This copyrighted material is made available to anyone wishing to use,
+# modify, copy, or redistribute it subject to the terms and conditions of
+# the GNU General Public License v.2, or (at your option) any later version.
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY expressed or implied, including the implied warranties of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General
+# Public License for more details.  You should have received a copy of the
+# GNU General Public License along with this program; if not, write to the
+# Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+# 02110-1301, USA.  Any Red Hat trademarks that are incorporated in the
+# source code or documentation are not subject to the GNU General Public
+# License and may only be used or replicated with the express permission of
+# Red Hat, Inc.
+#
+# Red Hat Author(s): Martin Kolman <mkolman@redhat.com>
+
+import os
+import unittest
+from unittest.mock import Mock
+import tempfile
+
+from dasbus.typing import *  # pylint: disable=wildcard-import
+from dasbus.error import DBusError
+
+from pyanaconda.modules.subscription.system_purpose import get_valid_fields, _normalize_field, \
+    _match_field, process_field, give_the_system_purpose
+
+# content of a valid populated valid values json file for system purpose testing
+SYSPURPOSE_VALID_VALUES_JSON = """
+{
+"role" : ["role_a", "role_b", "role_c"],
+"service_level_agreement" : ["sla_a", "sla_b", "sla_c"],
+"usage" : ["usage_a", "usage_b", "usage_c"]
+}
+"""
+
+# content of a valid but not populated valid values json file for system purpose testing
+SYSPURPOSE_VALID_VALUES_JSON_EMPTY = """
+{
+"role" : [],
+"service_level_agreement" : [],
+"usage" : []
+}
+"""
+
+
+class SystemPurposeLibraryTestCase(unittest.TestCase):
+    """Test the system purpose data handling code."""
+
+    def test_system_purpose_valid_json_parsing(self):
+        """Test that the JSON file holding valid system purpose values is parsed correctly."""
+        # check file missing completely
+        # - use path in tempdir to a file that has not been created and thus does not exist
+        with tempfile.TemporaryDirectory() as tempdir:
+            no_file = os.path.join(tempdir, "foo.json")
+            roles, slas, usage_types = get_valid_fields(valid_fields_file_path=no_file)
+            self.assertListEqual(roles, [])
```

Comment:
Please, drop this commit and create a new one with the upstream code. All `self.assert` calls were replaced with the `assert` statement.

Reply:
Ooops, now I see it. :P Should be fixed now. :)